kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RawCustomResourceOperationsImpl.java

```diff
@@ -696,15 +704,6 @@ public Watch watch(String namespace, String name, Map<String, String> labels, St
     return retVal;
   }
 
-  private String fetchUrl(String name, String namespace, Map<String, String> labels) {
-    String url = fetchUrl(namespace, labels);
-    if (name != null) {
-      return url + name;
-    } else {
-      return url.substring(0, url.length() - 1);
-    }
-  }
```

Comment:
Is it possible to keep the constructor clean and leave the additional logic here?
e.g.
```
  private String fetchUrl(String name, String namespace, Map<String, String> labels, String resourceVersion) {
    String resourceUrl = fetchUrl(namespace, labels);
    if (resourceUrl.endsWith("/")) {
      resourceUrl = resourceUrl.substring(0, resourceUrl.length() - 1);
    }
    URL url = new URL(resourceUrl);
    if (name != null) {
      httpUrlBuilder.addQueryParameter("fieldSelector", "metadata.name=" + name);
    }
    if (resourceVersion != null) {
      httpUrlBuilder.addQueryParameter("resourceVersion", resourceVersion);
    }
    return url;
  }
```

If you package expose the method, you could even add a couple of unit tests to verify the expected logic for each input variable combination.